js-core/homeworks/homework-17/src/main.js

```diff
@@ -45,9 +45,31 @@ class Carousel {
         this.transition = element.transition,
         this.infinity = element.infinity,
         this.nav = element.nav,
-        this.timer = element.timer
+        this.timer = element.timer,
+        this.list = carousel.querySelectorAll('li');
     }
-    start() {
+    static initialize(properties) {
+        let myInitializedCarousel =  new Carousel(properties);
+        myInitializedCarousel.start();
+    }
+    infinityCarousel () {
+        let last = this.list.length - 1;
+        if (!this.list[last].classList.contains('active')) {
+            for (let i = 0; i < this.list.length; i ++) {
+                let nextPosition = parseInt(this.list[i].style.left) - this.widthImg * this.count;
+                this.list[i].style.left = nextPosition + 'px';
+                let position = parseInt(this.list[i].style.left);
+                if (position === 0) {
+                    this.list[i].classList.add('active')
+                } else {
+                    this.list[i].classList.remove('active')
+                }
+            }
+        } else if (this.list[last].classList.contains('active') && this.infinity === true) {
```

Comment:
`this.list[last].classList.contains('active') && this.infinity === true`
well, having such conditionals is a tech-debt.

We can ask user for set default active element for example to avoid such construction or something else